packages/medusa/src/api/admin/return-reasons/[id]/route.ts

```diff
@@ -2,31 +2,38 @@ import {
   deleteReturnReasonsWorkflow,
   updateReturnReasonsWorkflow,
 } from "@medusajs/core-flows"
-import { UpdateOrderReturnReasonDTO } from "@medusajs/types"
+import {
+  AdminReturnReasonResponse,
+  UpdateOrderReturnReasonDTO,
+} from "@medusajs/types"
 import {
   ContainerRegistrationKeys,
+  MedusaError,
   remoteQueryObjectFromString,
 } from "@medusajs/utils"
 import {
   AuthenticatedMedusaRequest,
   MedusaResponse,
 } from "../../../../types/routing"
+import { refetchEntity } from "../../../utils/refetch-entity"
 
 export const GET = async (
   req: AuthenticatedMedusaRequest,
-  res: MedusaResponse
+  res: MedusaResponse<AdminReturnReasonResponse>
 ) => {
-  const remoteQuery = req.scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
-
-  const variables = { id: req.params.id }
-
-  const queryObject = remoteQueryObjectFromString({
-    entryPoint: "return_reason",
-    variables,
-    fields: req.remoteQueryConfig.fields,
-  })
-
-  const [return_reason] = await remoteQuery(queryObject)
+  const return_reason = await refetchEntity(
+    "return_reason",
+    req.params.id,
+    req.scope,
+    req.remoteQueryConfig.fields
+  )
+
+  if (!return_reason) {
```

Comment:
FIY:
`return await remoteQuery(queryObject, undefined, { throwIfKeyNotFound: true })`
this option in the remote query will throw if the entry is not found.

as we have multiple places using this refetchEntity, maybe it makes sense to add this parameter in the helper to avoid this same check on every "retrieve" endpoint